tidy3d/components/simulation.py

```diff
@@ -3777,23 +3777,26 @@ def _check_bloch_vec(
         if bloch_vec != expected_bloch_vec:
             test_val = np.real(expected_bloch_vec - bloch_vec)
 
-            if np.isclose(test_val % 1, 0) and not np.isclose(test_val, 0):
-                # the given Bloch vector is offset by an integer
-                log.warning(
-                    f"The wave vector of source '{source.name}' along dimension "
-                    f"'{dim}' is equal to the Bloch vector of the simulation "
-                    "boundaries along that dimension plus an integer reciprocal "
-                    "lattice vector. If using a 'DiffractionMonitor', diffraction "
-                    "order 0 will not correspond to the angle of propagation "
-                    "of the source. Consider using 'BlochBoundary.from_source()'.",
-                    custom_loc=["boundary_spec", "xyz"[dim]],
-                )
-            elif not np.isclose(test_val % 1, 0):
+            test_val_is_int = np.isclose(test_val, np.round(test_val))
+            src_name = f" '{source.name}'" if source.name else ""
+
+            # if test_val_is_int and not np.isclose(test_val, 0):
+            #     # the given Bloch vector is offset by an integer
+            #     log.warning(
+            #         f"The wave vector of source{src_name} along dimension "
+            #         f"'{dim}' is equal to the Bloch vector of the simulation "
+            #         "boundaries along that dimension plus an integer reciprocal "
+            #         "lattice vector. If using a 'DiffractionMonitor', diffraction "
+            #         "order 0 will not correspond to the angle of propagation "
+            #         "of the source. Consider using 'BlochBoundary.from_source()'.",
+            #         custom_loc=["boundary_spec", "xyz"[dim]],
+            #     )
+            if not test_val_is_int:
                 # the given Bloch vector is neither equal to the expected value, nor
                 # off by an integer
                 log.warning(
                     f"The Bloch vector along dimension '{dim}' may be incorrectly "
-                    f"set with respect to the source '{source.name}'. The absolute "
+                    f"set with respect to the source{src_name}. The absolute "
```

Comment:
Missing space

Reply:
The space is actually contained in `src_name`. The reason I made this variable was because otherwise the error message contains "source None" unless the user specifies a source name. So the space was added to handle both cases properly. 

```py
src_name = f" '{source.name}'" if source.name else ""
```